File: clients/ruby/model/player.rb
module Model

# Game's participant (team of units)
class Player
    # Unique id
    attr_accessor :id
    # Number of kills
    attr_accessor :kills
    # Total damage dealt to enemies
    attr_accessor :damage
    # Survival place (number of survivor teams currently/at the moment of death)
    attr_accessor :place
    # Team score
    attr_accessor :score

    def initialize(id, kills, damage, place, score)
        @id = id
        @kills = kills
        @damage = damage
        @place = place
        @score = score
    end

    # Read Player from input stream
    def self.read_from(stream)
        id = stream.read_int()
        kills = stream.read_int()
        damage = stream.read_double()
        place = stream.read_int()
        score = stream.read_double()
        Player.new(id, kills, damage, place, score)
    end

    # Write Player to output stream
    def write_to(stream)
        stream.write_int(@id)
        stream.write_int(@kills)
        stream.write_double(@damage)
        stream.write_int(@place)
        stream.write_double(@score)
    end

    def to_s
        string_result = "Player { "
        string_result += "id: "
        string_result += @id.to_s
        string_result += ", "
        string_result += "kills: "
        string_result += @kills.to_s
        string_result += ", "
        string_result += "damage: "
        string_result += @damage.to_s
        string_result += ", "
        string_result += "place: "
        string_result += @place.to_s
        string_result += ", "
        string_result += "score: "
        string_result += @score.to_s
        string_result += " }"
        string_result
    end

    def to_str
        to_s
    end
end

end

File: clients/javascript/model/player.js
/**
 * Game's participant (team of units)
 */
class Player {
    /**
     * Unique id
     */
    id;
    /**
     * Number of kills
     */
    kills;
    /**
     * Total damage dealt to enemies
     */
    damage;
    /**
     * Survival place (number of survivor teams currently/at the moment of death)
     */
    place;
    /**
     * Team score
     */
    score;

    constructor(id, kills, damage, place, score) {
        this.id = id;
        this.kills = kills;
        this.damage = damage;
        this.place = place;
        this.score = score;
    }

    /**
     * Read Player from input stream
     */
    static async readFrom(stream) {
        let id;
        id = await stream.readInt();
        let kills;
        kills = await stream.readInt();
        let damage;
        damage = await stream.readDouble();
        let place;
        place = await stream.readInt();
        let score;
        score = await stream.readDouble();
        return new Player(id, kills, damage, place, score);
    }

    /**
     * Write Player to output stream
     */
    async writeTo(stream) {
        let id = this.id;
        await stream.writeInt(id);
        let kills = this.kills;
        await stream.writeInt(kills);
        let damage = this.damage;
        await stream.writeDouble(damage);
        let place = this.place;
        await stream.writeInt(place);
        let score = this.score;
        await stream.writeDouble(score);
    }
}
module.exports = Player